### PR TITLE
fix: fix validation sorting false positives

### PR DESCRIPTION
**Pull request by rephus**
Closes: #5567
Closes: #5572

### Description:

Before: 
Screenshot from 2023-05-25 10-41-07

AFter:

Screenshot from 2023-05-25 10-41-48

### CODE REVIEW

File: packages/backend/src/services/ValidationService/ValidationService.ts [hunk 1]
```diff
@@ -200,25 +205,26 @@ export class ValidationService {
...
+                            acc,
+                            existingFieldIds,
+                            field.fieldId,
+                            `Sorting error: the field '${field.fieldId}' no longer exists`,
+                        ),
+                    [],
+                );
 
             /*
             // I think these are redundant, as we already check dimension/metrics
```
Comment (joaoviana) [inline packages/backend/src/services/ValidationService/ValidationService.ts:226]: should we just remove all the commented code below?

**Review by joaoviana — APPROVED** [review]

File: packages/backend/src/services/ValidationService/ValidationService.ts (hunk 1 above)
Comment (rephus) [reply packages/backend/src/services/ValidationService/ValidationService.ts:226]: not for now... I feel we might want to add it anyway, and it took me a while to generate those, specially the one about table calculations. For example, the method to filter table calculations was commented out and I have just recovered it.

Comment (rephus) [reply packages/backend/src/services/ValidationService/ValidationService.ts:226]: if you prefer, I can remove it and add a comment pointing to github, so we can easily track the code

Comment (joaoviana) [reply packages/backend/src/services/ValidationService/ValidationService.ts:226]: Cool, keep it @rephus, as long as we clean it up closer to the milestone completion then we're good! No action then, 👍

**Pull request merged**